server/src/com/thoughtworks/go/server/cache/GoCache.java

```diff
@@ -200,14 +200,20 @@ public void removeAll(List<String> keys) {
         }
     }
 
-    public void removeCompositeKeyFromParentCache(String key) {
-        if (key.contains(SUB_KEY_DELIMITER)) {
+    public void removeAssociations(String key, Element element) {
+        if (element.getValue() instanceof KeyList) {
+            synchronized (key.intern()) {
+                for (String subkey : (KeyList) element.getValue()) {
+                    remove(compositeKey(key, subkey));
```

Comment:
I'll probably merge this in now since it makes sense and is needed, but we could look later at [pulling this part out](https://github.com/jyotisingh/gocd/blob/69747ef2bdbca794a4fad15fdc292723e47352fe/server/src/com/thoughtworks/go/server/cache/GoCache.java#L171-L175) into its own method and using it here.